### PR TITLE
change all IO operation in legacy convert to context operation

### PR DESCRIPTION
**Pull request by 928PJY**

### CODE REVIEW

File: src/docfx/docset/Context.cs [hunk 1]
```diff
@@ -47,6 +47,11 @@ public void Report(Error error)
             _report.Write(error);
         }
 
+        public string ConvertToAbsolutePath(string relativePath)
```
Comment (herohua) [inline src/docfx/docset/Context.cs]: >ConvertToAbsolutePath [](start = 22, length = 21)

add a comment

File: src/docfx/docset/Context.cs [hunk 2]
```diff
@@ -47,6 +47,15 @@ public void Report(Error error)
             _report.Write(error);
         }
 
+        /// <summary>
+        /// Convert relative path to absolute path
+        /// </summary>
+        /// <param name="relativePath">Relatived to _outputPath</param>
+        public string ConvertToAbsolutePath(string relativePath)
```
Comment (OsmondJiang) [inline src/docfx/docset/Context.cs]: >ConvertToAbsolutePath [](start = 22, length = 21)

doesn't make sense to put this method to Context, would we put it to Config.Output?

Comment (yufeih) [reply src/docfx/docset/Context.cs]: or just use Path.Combine in legacy

**Comment by yufeih** [comment]
👏

File: src/docfx/build/legacy/LegacyExtensions.cs [hunk 3]
```diff
@@ -39,5 +39,10 @@ public static string ToLegacyOutputPath(this LegacyManifestOutputItem legacyMani
         {
             return Path.Combine(docset.Config.SiteBasePath ?? string.Empty, legacyManifestOutputItem.OutputPathRelativeToSiteBasePath);
         }
+
+        public static string ToAbsoluteOutputPath(this string relativePath, Docset docset)
```
Comment (OsmondJiang) [inline src/docfx/build/legacy/LegacyExtensions.cs]: >relativePath [](start = 62, length = 12)

all string will has this extension

Comment (928PJY) [reply src/docfx/build/legacy/LegacyExtensions.cs]: any advice?

---
In reply to: [199389546](https://github.com/dotnet/docfx/pull/2979#discussion_r199389546) [](ancestors = 199389546)

**Pull request merged**